src/main/scala/viper/gobra/translator/encodings/closures/ClosureSpecsEncoder.scala

```diff
@@ -81,6 +83,31 @@ protected class ClosureSpecsEncoder {
     } yield exp
   }
 
+  def goClosureCall(c: in.GoClosureCall)(ctx: Context): CodeWriter[vpr.Stmt] = {
+    import viper.silver.verifier.{errors => err}
+
+    register(c.spec)(ctx, c.spec.info)
+    val (pos, info, errT) = c.vprMeta
+    for {
+      member <- fromMemberLevel(callableMemberWithClosure(c.spec)(ctx))
+      args = closureCallArgs(c.closure, c.args, c.spec)(ctx)
+      vprargs <- sequence(args map ctx.expression)
+      (pres, formalParamDecls) = member match {
+        case f: vpr.Function => (f.pres, f.formalArgs)
+        case m: vpr.Method => (m.pres, m.formalArgs)
+        case e => Violation.violation(s"expected a function or method, but found $e instead.")
+      }
+      formalParams = formalParamDecls.map(_.localVar)
+      substitutions = (formalParams zip vprargs).toMap
+      instantiatedPres = pres.map(_.replace(substitutions))
+      and = ViperUtil.bigAnd(instantiatedPres)(pos, info, errT)
+      exhale = vpr.Exhale(and)(pos, info, errT)
+      _ <- errorT {
+        case err.ExhaleFailed(Source(info), _, _) => PreconditionError(info).dueTo(GoCallPreconditionReason(info))
+      }
```

Comment:
I believe the code writer has a method to generate an exhale together with adding an error transformation. I recommend using that one to avoid errors where the error transformer is hocked incorrectly. That said, I am also fine if you prefer this style.